ts/query_test.go

```diff
@@ -224,9 +266,16 @@ func (tm *testModel) assertQuery(name string, r Resolution, start, end int64,
 	}
 	iters.init()
 	for iters.isValid() {
+		var value float64
+		switch q.GetAggregator() {
+		case proto.TimeSeriesQueryAggregator_AVG:
+			value = iters.avg()
+		case proto.TimeSeriesQueryAggregator_AVG_RATE:
+			value = iters.dAvg()
+		}
```

Comment:
Might be cleaner/faster to pull the switch out of the loop and store `iters.avg/iters.dAvg` in a function pointer:

```
  var value func() float64
  switch q.GetAggregator() {
    case proto.TimeSeriesQueryAggregator_AVG:
      value = iters.avg
    case proto.TimeSeriesQueryAggregator_AVG_RATE:
      value = iters.dAvg
  }
  for iters.isValid() {
    expectedDataPointers = append(expectedDatapoints, &proto.TimeSeriesDatapoint{
      TimestampNanos: iters.timestamp(),
      Value: value(),
    })
    iters.advance()
  }
```